orchagent/aclorch.cpp

```diff
@@ -1780,6 +1860,64 @@ void AclOrch::init(vector<TableConnector>& connectors, PortsOrch *portOrch, Mirr
 {
     SWSS_LOG_ENTER();
 
+    // TODO: Query SAI to get mirror table capabilities
+    // Right now, verified platforms that support mirroring IPv6 packets are
+    // Broadcom and Mellanox. Virtual switch is also supported for testing
+    // purposes.
+    string platform = getenv("platform") ? getenv("platform") : "";
+    if (platform == BRCM_PLATFORM_SUBSTRING ||
+            platform == MLNX_PLATFORM_SUBSTRING)
+    {
+        m_mirrorTableCapabilities =
+        {
+            { ACL_TABLE_MIRROR, true },
+            { ACL_TABLE_MIRRORV6, true },
+        };
+    }
+    else
+    {
+        m_mirrorTableCapabilities =
+        {
+            { ACL_TABLE_MIRROR, true },
+            { ACL_TABLE_MIRRORV6, false },
+        };
+    }
+
+    SWSS_LOG_NOTICE("%s switch capability:", platform.c_str());
+    SWSS_LOG_NOTICE("    ACL_TABLE_MIRROR: %s",
+            m_mirrorTableCapabilities[ACL_TABLE_MIRROR] ? "yes" : "no");
+    SWSS_LOG_NOTICE("    ACL_TABLE_MIRRORV6: %s",
+            m_mirrorTableCapabilities[ACL_TABLE_MIRRORV6] ? "yes" : "no");
+
+    // In Broadcom platform, V4 and V6 rules are stored in the same table
+    if (platform == BRCM_PLATFORM_SUBSTRING) {
+        m_isCombinedMirrorV6Table = true;
+    }
+
+    // In Mellanox platform, V4 and V6 rules are stored in different tables
+    if (platform == MLNX_PLATFORM_SUBSTRING) {
+        m_isCombinedMirrorV6Table = false;
+    }
+
+    // Store the capabilities in state database
+    vector<FieldValueTuple> fvVector;
+    for (auto const& it : m_mirrorTableCapabilities)
+    {
+        string value = it.second ? "true" : "false";
+        switch (it.first)
+        {
+            case ACL_TABLE_MIRROR:
+                fvVector.emplace_back(TABLE_TYPE_MIRROR, value);
+                break;
+            case ACL_TABLE_MIRRORV6:
+                fvVector.emplace_back(TABLE_TYPE_MIRRORV6, value);
+                break;
+            default:
+                break;
+        }
+    }
+    m_switchTable.set("switch", fvVector);
```

Comment:
this code really need to go to syncd code. We will address this in the next release.

Reply:
sure i could add some todo to here

Reply:
One issue about moving this to syncd right now is that we currently cannot guarantee the order of syncd and orchagent starting time. e.g. when orchagent starts and at that time syncd hasn't stored the capability information into the database, we will miss this part of the information.